### ISSUE TITLE
[ci/python] Run full CI on merges to `release-i.j` branches

### ISSUE THREAD

**Issue by johnkerl** [single-cell-data/TileDB-SOMA#1489]
## Problem 1

Today we had a CI fail (on MacOS and Python 3.7 only) once 1.2.7 was tagged:
https://github.com/single-cell-data/TileDB-SOMA/actions/runs/5336828062/jobs/9672650908

This may be because GithHub just updated the runner images yesterday: https://github.com/actions/runner-images/pull/7752

Regardless, this needs to get caught pre-tag not post-tag.

We run "full" CI on merges to `main`, and on releases (post-tag):
https://github.com/single-cell-data/TileDB-SOMA/blob/main/.github/workflows/python-ci-full.yml#L6-L10

That was mostly a good decision -- "full CI" is distinct from "minimal CI" in that the latter runs on every PR and we were running into resource-starvation issues which were keeping people from doing dev work: https://github.com/single-cell-data/TileDB-SOMA/issues/693

We need to modify `python-ci-full.yml` to run not only on merges to `main` and on releases (post-tag) but also on merges to any `release-i.j` branch.

## Problem 2

Issue here: https://github.com/actions/setup-python/issues/682

affects us here:
https://github.com/single-cell-data/TileDB-SOMA/actions/runs/5336828062/jobs/9672070106#step:8:40

That was today. I went back and re-ran a successful CI run from last week, and on that re-run today, it failed as above:
https://github.com/single-cell-data/TileDB-SOMA/actions/runs/5269863811/jobs/9673256042

Per @ihnorton : we can use https://github.com/PySport/kloppy/commit/16079581a6a03b46fada13324d52b8da9f8550b1 to get our `.github/worksflows` YAML using non-broken runneers

**Comment by johnkerl** [comment]
@mlin @thetorpedodog I tried this out on
https://github.com/single-cell-data/TileDB-SOMA/pull/1492
which was a PR merged to `release-1.2` but at
https://github.com/single-cell-data/TileDB-SOMA/actions/workflows/python-ci-full.yml
I did _not_ get a full CI run when #1492 was merged to `release-1.2`

Do you have any advice?

**Comment by thetorpedodog** [comment]
> I notice that for core they use no quotes on `release-*`: https://github.com/TileDB-Inc/TileDB/blob/2.15.4/.github/workflows/full-ci.yml#L6

The quotes shouldn’t make a difference; either way YAML will parse it as the string `release-*` (just that the quoted version is explicit that it’s a string)[^0]; i.e., both of these blocks are equivalent to the JSON object `{"branches": ["main", "release-*"]}`:

```yaml
branches:
  - main
  - "release-*"

branches:
  - "main"
  - release-*
```

[^0]: i do not like yaml

**Comment by thetorpedodog** [comment]
I figured it out: the workflows that are run are based on the contents of the `.github` directory *at the revision which is pushed*; they’re not global from what happens in `main`. So because the `.github/...` directory *on the release-1.2 branch* didn’t match `release-1.2`, there was nothing to run. This will be fixed by #1497, and in the future we will be covered by the fact that when we branch off `release-1.3` and beyond, they will include the `release-*` pattern in their `.github` workflow file.

**Comment by johnkerl** [comment]
D'oh ;) 

Thanks @thetorpedodog !!!

**Comment by johnkerl** [comment]
Resolved by #1497:
https://github.com/single-cell-data/TileDB-SOMA/actions/workflows/python-ci-full.yml?query=branch%3Arelease-1.2
<img width="974" alt="Screenshot 2023-06-21 at 10 00 23 PM" src="https://github.com/single-cell-data/TileDB-SOMA/assets/2008794/42e611b7-7599-479d-827c-a2d61d9e6b93">
